### PR TITLE
redo-apenwarr: switch to python3 and fix building man

### PR DESCRIPTION
**Pull request by ck3d**
removed dependency to mkdoc, which is only needed to build the webpage.
Python3 is not supported in 0.42, so I had to update to current master.

###### Motivation for this change
switch to python3

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/build-managers/redo-apenwarr/default.nix [hunk 1]
```diff
@@ -1,27 +1,16 @@
...
-
-  mkdocs-exclude = python27.pkgs.callPackage ./mkdocs-exclude.nix {
-    pythonPackages = python27.pkgs;
-  };
-in stdenv.mkDerivation rec {
+}: stdenv.mkDerivation rec {
 
   pname = "redo-apenwarr";
-  version = "0.42";
+  version = "20200304";
```
Comment (markuskowa) [inline pkgs/development/tools/build-managers/redo-apenwarr/default.nix]: version [0.42a](https://github.com/apenwarr/redo/releases) has just been released. Maybe this is better than going to unstable?

Comment (ck3d) [reply pkgs/development/tools/build-managers/redo-apenwarr/default.nix]: What a coincidence. Thanks, I updated the version, sha256 is still the same.

**Comment by markuskowa** [comment]
Backported to 20.03 in bdd53db9421b98d5593b69356560880109809a9a
(ZHF https://github.com/NixOS/nixpkgs/issues/80379)

**Comment by danderson** [comment]
We raced each other! I was about to send the same PR :).

Just for the record: upstream released 0.42a specifically for python3 support, so the upgrade here is not to HEAD per se, but to the next official upstream release.

Thanks for fixing!

**Pull request merged**